### PR TITLE
Add support for External Groups

### PR DESCRIPTION
**Pull request by kabesan**
Fixes: #2213 .
https://docs.github.com/en/enterprise-cloud@latest/rest/reference/teams#external-groups

### CODE REVIEW

File: github/teams.go
Comment (gmlewis) [inline github/teams.go:957]:
```suggestion
// RemoveConnectedExternalGroup removes the connection between an external group and a team.
```

File: github/teams_test.go
Comment (gmlewis) [inline github/teams_test.go:1830]: Please add a blank line before this test.

File: github/teams_test.go
Comment (gmlewis) [inline github/teams_test.go:1905]: Please add a blank line before this test.

File: github/teams_test.go
Comment (gmlewis) [inline github/teams_test.go:2020]: Please add a blank line before this test.

File: github/teams_test.go
Comment (gmlewis) [inline github/teams_test.go:2074]: Please add a blank line before this test.

**Review by gmlewis — CHANGES_REQUESTED** [review]
Thank you, @kabesan !
Just a few tweaks, please, then we should be ready for a second LGTM before merging.

**Comment by gmlewis** [comment]
Also, please make sure to run unit tests before pushing (not force-pushing) your upcoming changes.
We squash and merge, so don't worry about the number of commits.
And finally, make sure to run `go generate ./...` and push those changes.
See our CONTRIBUTING.md file for more details.

**Comment by kabesan** [comment]
@gmlewis Thank your review.
I fiexed, Please re-review.

File: github/teams_test.go
Comment (gmlewis) [inline github/teams_test.go:1793]: Here, and everywhere else you use `time.Parse`, please change this to use the already-existing `referenceTime` value created in: `github/timestamp_test.go`.

File: github/teams.go
Comment (gmlewis) [inline github/teams.go:876]: Please run this file through `go fmt` to fix the indentation.

**Review by gmlewis — CHANGES_REQUESTED** [review]

**Review by gmlewis — APPROVED** [review]
Thank you, @kabesan !
LGTM.

Awaiting second LGTM before merging.

(Please note that ALL other contributors to this repo are welcome to provide the second PR review/comment/approval that we need for merging and that we are not waiting for any particular reviewer unless otherwise noted.)

**Comment by kabesan** [comment]
Fixed for diff coverage!

**Review by gmlewis — APPROVED** [review]

**Comment by kabesan** [comment]
@gmlewis Hi
Is second reviewer not active?
Can you add other reviewers?

**Comment by gmlewis** [comment]
> @gmlewis Hi Is second reviewer not active? Can you add other reviewers?

I am no longer able to add other reviewers, but any other contributor to this repo is welcome to review.

**Comment by kabesan** [comment]
Hi @pierce-m
Can you review this PR?
I'm waiting for second review.

**Comment by pierce-m** [comment]
@kabesan Yeah I'll take a look, but probably won't be able to until after the winter holidays

**Review by pierce-m — APPROVED** [review]

**Comment by gmlewis** [comment]
Thank you, @pierce-m !
Merging.

**Pull request merged**
